lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -941,6 +941,144 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+        }
+        for (int64_t i : strides) {
+          Value x = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(i));
+          cstStrides.push_back(x);
+        }
+
+        Value zero = rewriter.create<Torch::ConstantIntOp>(
+            binder.getLoc(), rewriter.getI64IntegerAttr(0));
+
+        // Index the imageShape and blockShape tensors, as AtenCol2imOp expects
+        // them to be int lists.
+        auto select = [&](Value v, Value k,
+                          Torch::ValueTensorType ty) -> Value {
+          Value kTensor = rewriter.create<Torch::PrimNumToTensorScalarOp>(
+              binder.getLoc(),
+              Torch::ValueTensorType::get(
+                  binder.op->getContext(), ArrayRef<int64_t>{1},
+                  rewriter.getIntegerType(64, /*signed*/ 1)),
+              k);
+
+          auto sel = rewriter.create<Torch::AtenIndexSelectOp>(
+              binder.getLoc(),
+              Torch::ValueTensorType::get(ty.getContext(), ArrayRef<int64_t>{1},
+                                          ty.getOptionalDtype()),
+              v, zero, kTensor);
+          Value item = rewriter.create<Torch::AtenItemOp>(
+              binder.getLoc(), rewriter.getType<Torch::IntType>(), sel);
+          return item;
+        };
+
+        SmallVector<Value> imageShapeContainer, blockShapeContainer;
+        for (int64_t i = 0; i < imageShapeSizes[0]; ++i) {
+          Value k = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(i));
+
+          // Passing in the shapeType of each of these tensors avoids
+          // repeated casts, as these have already been calculated.
+          imageShapeContainer.push_back(select(imageShape, k, imageShapeTy));
+          blockShapeContainer.push_back(select(blockShape, k, blockShapeTy));
+        }
+
+        Value imageShapeAsList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            imageShapeContainer);
+        Value blockShapeAsList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            blockShapeContainer);
+        Value paddingList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            cstPadding);
+        Value dilationsList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            cstDilations);
+        Value stridesList = rewriter.create<Torch::PrimListConstructOp>(
+            binder.getLoc(),
+            Torch::ListType::get(Torch::IntType::get(binder.op->getContext())),
+            cstStrides);
```

Comment:
Instead of doing this, and creating the constants above you can use `createConstantList` utility.

Reply:
Ah, I did not know that this utility existed. The code is much shorter now. Thanks!